Marlin/src/lcd/menu/menu_probe_offset.cpp

```diff
@@ -0,0 +1,158 @@
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+//
+// Calibrate Probe offset menu.
+//
+
+#include "../../inc/MarlinConfigPre.h"
+
+#if BOTH(HAS_LCD_MENU, PROBE_OFFSET_MENU) && DISABLED(BABYSTEP_ZPROBE_OFFSET)
+
+#include "menu.h"
+#include "menu_addon.h"
+#include "../../module/motion.h"
+#include "../../module/planner.h"
+
+#if HAS_LEVELING
+  #include "../../feature/bedlevel/bedlevel.h"
+#endif
+
+#if HAS_BED_PROBE 
+  #include "../../module/probe.h"
+#endif
+
+#ifndef PROBE_OFFSET_START 
+  #define PROBE_OFFSET_START -4.0
+#endif
+
+static_assert(PROBE_OFFSET_START < 0, "PROBE_OFFSET_START must be < 0. Please update your configuration.");
+
+#if HAS_LEVELING
+  static bool leveling_was_active = false;
+#endif
+
+#if HAS_SOFTWARE_ENDSTOPS
+  bool store_soft_endstops_enabled;
+#endif  
+float store_z_offset;
+float calculated_z_offset;
+
+static void _lcd_probe_offset_move_undo(){
+  #if HAS_SOFTWARE_ENDSTOPS
+    soft_endstops_enabled = store_soft_endstops_enabled;
+  #endif  
+  probe.offset.z = store_z_offset;
+  ui.goto_previous_screen_no_defer();
+  
+}
+
+static void _lcd_probe_offset_move_done(){
+  probe.offset.z = calculated_z_offset;
+  #if HAS_SOFTWARE_ENDSTOPS
+    soft_endstops_enabled = store_soft_endstops_enabled;
+  #endif
+
+  #if HAS_LEVELING
+    set_bed_leveling_enabled(leveling_was_active);
+  #endif
+  #if Z_AFTER_HOMING
+    do_z_clearance(Z_AFTER_HOMING);
+  #else
+    do_z_clearance(20.0);
+  #endif
+  
+  ui.goto_previous_screen_no_defer();
+};
+
+void _goto_manual_move_z(const float scale) {
+  ui.defer_status_screen();
+  ui.manual_move.menu_scale = scale;
+  ui.goto_screen(lcd_move_z);
+}
+
+static void _lcd_probe_offset_move_z() {
+  _lcd_draw_homing();
+  if (all_axes_homed()) {
+     ui.goto_screen([]{
+      ui.defer_status_screen();
+      START_MENU();
+      calculated_z_offset = probe.offset.z + current_position.z;
+
+      if (LCD_HEIGHT >= 4) {
+        STATIC_ITEM(MSG_MOVE_Z, SS_CENTER|SS_INVERT); 
+      }
+
+      VALUE_ITEM(MSG_ZPROBE_ZOFFSET, ftostr42_52(calculated_z_offset), SS_LEFT);
+
+      SUBMENU(MSG_MOVE_1MM,  []{ _goto_manual_move_z( 1);    });
+      SUBMENU(MSG_MOVE_01MM, []{ _goto_manual_move_z( 0.1f); });
```

Comment:
Probe offset is not so high value.
Don't you want to leave 0.01 step only? 
By analogy with Probe Z Offset menu I mean.

Reply:
If you are configured like me, after homing Z moves to 20mm, you really dont want to move down 16 to 18mm using the .01 step with one of the usual rotary encoders fitted to most printers (like mine).